src/main/java/org/jenkinsci/remoting/protocol/IOHub.java

```diff
@@ -83,6 +92,10 @@
      * Our selector.
      */
     private final Selector selector;
+    private volatile boolean winWatcherSelectorIsWaiting = false;
+    private volatile boolean ioHubRunning = false;
```

Comment:
Use a lock object so that the wakeup thread can use `ioHubRunningLock.wait(SELECTOR_WAKEUP_TIMEOUT_MS)` rather than a volatile... otherwise you will have to wait for the remainder of the sleep to shut down the executor